### PR TITLE
Enable workflow identity

### PR DESCRIPTION
**Pull request by wiktork**

### CODE REVIEW

File: src/Extensions/AzureBlobStorage/AzureBlobEgressProviderOptions.cs [hunk 1]
```diff
@@ -41,6 +41,11 @@ internal sealed partial class AzureBlobEgressProviderOptions
             Description = nameof(OptionsDisplayStrings.DisplayAttributeDescription_AzureBlobEgressProviderOptions_ManagedIdentityClientId))]
         public string ManagedIdentityClientId { get; set; }
 
+        [Display(
+            ResourceType = typeof(OptionsDisplayStrings),
+            Description = nameof(OptionsDisplayStrings.DisplayAttributeDescription_AzureBlobEgressProviderOptions_UseWorkloadIdentity))]
+        public bool UseWorkloadIdentity { get; set; }
```
Comment (jander-msft) [inline src/Extensions/AzureBlobStorage/AzureBlobEgressProviderOptions.cs:47]: Is this specific enough? Should it be named something like `UseWorkloadIdentityFromEnvironment`? Or if we added other properties in the future to allow specifying the credentials, would this property as need to be set to `true`?

File: src/Extensions/AzureBlobStorage/AzureBlobEgressProvider.cs [hunk 2]
```diff
@@ -375,22 +376,39 @@ private static string WrapMessage(string innerMessage)
...
+            if (options.UseWorkloadIdentity)
+            {
+                credOptions.ExcludeWorkloadIdentityCredential = false;
+            }
+
+            if (!string.IsNullOrEmpty(options.ManagedIdentityClientId))
+            {
+                credOptions.ExcludeManagedIdentityCredential = false;
+                credOptions.ManagedIdentityClientId = options.ManagedIdentityClientId;
+            }
```
[the remark this answers is not in the document]
Comment (wiktork) [reply src/Extensions/AzureBlobStorage/AzureBlobEgressProvider.cs:395]: Based on https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/identity/Azure.Identity/README.md, workload identity will take precedence.

[the remark this answers is not in the document]
Comment (wiktork) [reply src/Extensions/AzureBlobStorage/AzureBlobEgressProvider.cs:395]: Per your comment, will also add a test to ensure all new auth systems are disabled by default.

File: src/Extensions/AzureBlobStorage/AzureBlobEgressProviderOptions.cs (hunk 1 above)
Comment (wiktork) [reply src/Extensions/AzureBlobStorage/AzureBlobEgressProviderOptions.cs:47]: I didn't want to do this initially, but thinking about it some more if we add new properties i want to be able to compose them (i.e. bring in some environment variables and some from properties) so I will rename this.

**Review by jander-msft — APPROVED** [review]

**Pull request merged**